### PR TITLE
feat: add documents validation

### PR DESCRIPTION
**Pull request by pedrobonamin**
### Description

It introduces document validation using the preview store listener in the bundles details screen.
It shows in the UI (WIP) the documents with validation errors.

It also refactors document-store pair validation function by abstracting the part of that function that takes care of **validating** the document from the **fetcher**. 
The document-store pair uses the document-store to fetch the document, which makes it not scalable enough for this use case, by dividing the validation from the fetcher, we could provide a document to the function `validateDocumentWithReferences` and this function will take care of the validation. 

https://www.loom.com/share/0a027910d89844868206d2b54c824cb3?sid=84e86aaa-de88-46f2-9917-4b20ea14442b

### CODE REVIEW

File: packages/sanity/src/core/releases/tool/detail/review/DocumentReviewHeader.tsx [hunk 1]
```diff
@@ -1,67 +1,76 @@
...
-            text={
-              <span>
-                Created <RelativeTime time={document._createdAt} useTemporalPhrase />
-              </span>
-            }
+    <Card
+      tone={validation?.hasError ? 'critical' : 'default'}
+      radius={3}
+      style={{marginBottom: 20}}
+    >
```
Comment (pedrobonamin) [inline packages/sanity/src/core/releases/tool/detail/review/DocumentReviewHeader.tsx:35]: This was added to this component to wrap it and show the validation error, only change here.

File: packages/sanity/src/core/store/_legacy/document/document-pair/validation.ts [hunk 2]
```diff
@@ -114,81 +49,7 @@ export const validation = memoize(
       shareLatestWithRefCount(),
     )
 
-    const referenceIds$ = document$.pipe(
```
Comment (pedrobonamin) [inline packages/sanity/src/core/store/_legacy/document/document-pair/validation.ts:117]: all this logic below has been moved to [`validateDocumentWithReferences`](https://github.com/sanity-io/sanity/pull/7207/files#diff-f33b0c793316d3d8640c334231c4ecdef6eca59ba5b8b5bbab0698d46d33a906)

File: packages/sanity/src/core/validation/validateDocumentWithReferences.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
+import {type SanityClient} from '@sanity/client'
```
Comment (pedrobonamin) [inline packages/sanity/src/core/validation/validateDocumentWithReferences.ts:1]: All of this was previously handled by `document-pair/validation` 
It was moved here to make it possible to reuse by receiving `document$: Observable<SanityDocument>`

File: packages/sanity/src/core/releases/tool/detail/bundleDocumentsValidation.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+          ...validation,
+          documentId: id,
+          hasError: validation.validation.some((marker) => isValidationErrorMarker(marker)),
+        })),
+      )
+    }),
+  ).pipe(
+    // Transform to a dictionary with the id and value
+    scan((acc, next) => {
+      const newMap = new Map(acc) // Create a new Map instance based on the current accumulator
```
Comment (bjoerge) [inline packages/sanity/src/core/releases/tool/detail/bundleDocumentsValidation.ts:63]: Is there any benefit of using a `Map` here? I guess in theory we get faster lookups, but we're also instantiating new maps at each change and wondering if using plain objects would work just as well (we'd have to instantiate those too ofc, but might be cheaper?). I'm mostly curious, although we tend to be quite consistent about passing plain objects as React props, so diverging from that may complicate things slightly later on.

File: packages/sanity/src/core/releases/tool/detail/useBundleDocumentsValidation.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  const observable = useMemo(() => {
+    return bundleDocumentsValidation(
+      {
+        observeDocumentPairAvailability: unstable_observeDocumentPairAvailability,
+        observeDocument: unstable_observeDocument,
+        schema,
+        i18n,
+        getClient,
+      },
+      bundlesIds.split(','),
```
Comment (bjoerge) [inline packages/sanity/src/core/releases/tool/detail/useBundleDocumentsValidation.ts:25]: Should make a note here that this is a hack and something that we should look into fixing later

File: packages/sanity/src/core/validation/validateDocumentWithReferences.ts (hunk 3 above)
Comment (bjoerge) [reply packages/sanity/src/core/validation/validateDocumentWithReferences.ts:1]: Very nice!

File: packages/sanity/src/core/releases/tool/detail/bundleDocumentsValidation.ts (hunk 4 above)
Comment (pedrobonamin) [reply packages/sanity/src/core/releases/tool/detail/bundleDocumentsValidation.ts:63]: Not really, I just prefer the map api vs plain objects when creating it for this type of use cases. Given the size of the object or map is quite small I don't think there will really be a performance difference when accessing it. 
Will change to an object for consistency

Comment (pedrobonamin) [reply packages/sanity/src/core/releases/tool/detail/bundleDocumentsValidation.ts:63]: did the same change for document history, which was also using a map.

**Review by bjoerge — APPROVED** [review]
Great work!

**Pull request merged**